xarray/core/variable.py

```diff
@@ -1949,6 +1949,11 @@ def _coarsen_reshape(self, windows, boundary, side):
             else:
                 shape.append(variable.shape[i])
 
+        keep_attrs = _get_keep_attrs(default=False)
+        if keep_attrs is None:
+            keep_attrs = _get_keep_attrs(default=False)
```

Comment:
```suggestion
        if keep_attrs is None:
            keep_attrs = _get_keep_attrs(default=False)
```

Reply:
If the assignment on L.1952 is removed, the `Variable.coarsen` tests fail with `UnboundLocalError: local variable 'keep_attrs' referenced before assignment`

Reply:
Ah yes; then we don't need the bottom two lines; just `keep_attrs = _get_keep_attrs(default=False)` is enough